StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace StyleCop.Analyzers.Test.SpacingRules
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using StyleCop.Analyzers.SpacingRules;
+    using TestHelper;
+    using Xunit;
+
+    /// <summary>
+    /// Unit tests for <see cref="SA1004DocumentationLinesMustBeginWithSingleSpace"/>.
+    /// </summary>
+    public class SA1004UnitTests : DiagnosticVerifier
+    {
+        /// <summary>
+        /// Verifies that the analyzer will properly handle an empty source.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestEmptySourceAsync()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFixedExampleAsync()
+        {
+            string testCode = @"
+public class TypeName
+{
+    /// <summary>
+    /// The summary text.
+    /// </summary>
+    /// <param name=""x"">The document root.</param>
+    /// <param name=""y"">The XML header token.</param>
+    private void Method1(int x, int y)
+    {
+    }
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestRuleExampleAsync()
+        {
+            string testCode = @"
+public class TypeName
+{
+    ///<summary>
+    ///The summary text.
+    ///</summary>
+    ///   <param name=""x"">The document root.</param>
+    ///    <param name=""y"">The XML header token.</param>
+    private void Method1(int x, int y)
+    {
+    }
+}
+";
+
+            // Currently the extra indentation for <param> elements is not checked.
```

Comment:
Do we *want* to check for extra indentation? It caused a problem in the past (#715). If you think we should check extra indentation for some set of rules, I'm thinking we should create a new issue to define exactly what those rules are.